pandas/io/common.py

```diff
@@ -232,13 +233,46 @@ def file_path_to_url(path):
 }
 
 
+def _get_compression_method(compression: Optional[Union[str, Dict[str, Any]]]):
```

Comment:
I think the Dict can be typed as `Dict[str, str]` here. Would also be nice to type the return though that may change depending on comment below

Reply:
can you do this